subprojects/dependency-lock/src/integTest/groovy/org/gradle/internal/dependencylock/generation/DependencyLockFileSubsequentBuildsIntegrationTest.groovy

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.dependencylock.generation
+
+import static org.gradle.internal.dependencylock.fixtures.DependencyLockFixture.*
+
+class DependencyLockFileSubsequentBuildsIntegrationTest extends AbstractDependencyLockFileGenerationIntegrationTest {
```

Comment:
We seem to be missing coverage for updating an existing lock file when a different configuration is resolved, or I just can't find it. This should probably include the case where the second configuration is a parent or child of the first resolved configuration.

Reply:
done